code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DefaultDeploymentUnitInstance.java

```diff
@@ -60,14 +56,6 @@ public void remove() {
                                 InstanceConstants.PROCESS_STOP, InstanceConstants.PROCESS_REMOVE));
             }
         }
-
-        // 2) remove the mapping
-        List<? extends ServiceExposeMap> maps = context.objectManager.mappedChildren(
```

Comment:
In the abstract class, we're just scheduling for removal one ServiceExposeMap (the class member exposeMap).  But here, we were removing a List of them (all of the ones associated with this instance).